File: solutions/c++/restaurantmanagementsystem/OrderItem.cpp
#include "OrderItem.hpp"
#include <iostream>
#include <iomanip>

OrderItem::OrderItem(MenuItem* menuItem, int quantity)
    : menuItem(menuItem), quantity(quantity) {
    subtotal = menuItem->getPrice() * quantity;
}

MenuItem* OrderItem::getMenuItem() const { return menuItem; }
int OrderItem::getQuantity() const { return quantity; }
double OrderItem::getSubtotal() const { return subtotal; }

void OrderItem::updateQuantity(int quantity) {
    if (quantity > 0) {
        this->quantity = quantity;
        subtotal = menuItem->getPrice() * quantity;
    }
}

void OrderItem::displayInfo() const {
    std::cout << menuItem->getName() << " x " << quantity 
              << " = $" << std::fixed << std::setprecision(2) << subtotal << std::endl;
} 

File: solutions/java/src/restaurantmanagementsystem/OrderItem.java
package restaurantmanagementsystem;

public class OrderItem {
    private final MenuItem item;
    private final int quantity;

    public OrderItem(MenuItem item, int quantity) {
        this.item = item;
        this.quantity = quantity;
    }

    public double getSubtotal() {
        return item.getPrice() * quantity;
    }

    public MenuItem getItem() { return item; }
}
